services/database/scripts/migrateUserStates.js

```diff
@@ -0,0 +1,210 @@
+
+  if (states === undefined) {
+    // This user has already been migrated; no update necessary.
+    return false;
+  }
+
+  if (role === "state" && Array.isArray(states)) {
+    user.state = user.states[0];
+  }
+
+  // If we have a more trustworthy date, use it. Otherwise, leave what's there.
+  user.lastLogin = loginDates.get(usernameSub) ?? user.lastLogin;
+
+  delete user.states;
+  delete user.isSuperUser;
+  delete user.password;
+  return true;
+}
+
+/**
+ * Create an iterator for the entire table.
+ * @param {string} TableName 
+ */
+async function * scanAllUsers (TableName) {
+  console.log(`${logPrefix()}Scanning...`);
+  let pageNumber = 0;
+  let totalItemCount = 0;
+  for await (let page of paginateScan({ client }, { TableName })) {
+    pageNumber += 1;
+    let pageItemCount = 0;
+    for (let user of (page.Items ?? [])) {
+      pageItemCount += 1;
+      totalItemCount += 1;
+      yield user;
+    }
+    console.log(`${logPrefix()}Completed scan of page ${pageNumber}; ${pageItemCount} items processed.`);
+  }
+  console.log(`${logPrefix()}Scan complete; ${totalItemCount} items in all.`);
+}
+
+/**
+ * @param {AsyncGenerator<object>} userIterable 
+ */
+async function * filterAndModifyUsers (userIterable, loginDates) {
+  for await (let user of userIterable) {
+    const needsUpdate = updateUserState(user, loginDates);
+    if (needsUpdate) {
+      yield user;
+    }
+  }
+}
+
+/**
+ * @param {AsyncGenerator<object>} itemIterable
+ */
+async function * createBatches (itemIterable) {
+  let batch = [];
+
+  for await (let item of itemIterable) {
+    batch.push(item);
+    if (batch.length === 25) {
+      yield batch;
+      batch = [];
+    }
+  }
+
+  if (batch.length > 0) {
+    yield batch;
+  }
+}
+
+/**
+ * Send BatchWriteCommands to Put the given objects in the given table.
+ * @param {AsyncGenerator<object[]>} batches
+ * @param {string} tableName
+ * @param {(object) => string} idSelector For logging purposes.
+ */
+async function sendBatches (batches, tableName, idSelector) {
+  for (let batch of batches) {
+    const command = new BatchWriteCommand({
+      RequestItems: {
+        [tableName]: batch.map((Item) => ({
+          PutRequest: { Item }
+        }))
+      }
+    });
+    const response = await client.send(command);
+    const unprocessedItems = response.UnprocessedItems?.[tableName];
+    if (unprocessedItems && unprocessedItems.length > 0) {
+      const ids = unprocessedItems.map((putRequest) => idSelector(putRequest.Item));
+      throw new Error(
+        `Batch write failed! The following items were not updated: ${ids.join(", ")}`
+      );
+    }
+  }
+}
+
+async function main () {
+  const lastLoginDates = await gatherLoginDatesFromCognito();
+  const allUsers = scanAllUsers(AUTH_USER_TABLE_NAME);
+  const usersToUpdate = filterAndModifyUsers(allUsers, lastLoginDates);
+  const batches = createBatches(usersToUpdate);
+  await sendBatches(AUTH_USER_TABLE_NAME, batches, user => user.userId);
```

Comment:
the order of the parameters is flipped.
```suggestion
  await sendBatches(batches, AUTH_USER_TABLE_NAME, user => user.userId);
```